src/asynk/async_queue.rs

```diff
@@ -330,20 +340,21 @@ where
     <Tls as MakeTlsConnect<Socket>>::TlsConnect: Send,
     <<Tls as MakeTlsConnect<Socket>>::TlsConnect as TlsConnect<Socket>>::Future: Send,
 {
-    pub async fn connect(
-        uri: impl ToString,
-        tls: Tls,
-        duplicated_tasks: bool,
-    ) -> Result<Self, AsyncQueueError> {
-        let manager = PostgresConnectionManager::new_from_stringlike(uri, tls)?;
-        let pool = Pool::builder().build(manager).await?;
+    pub async fn connect(&mut self, tls: Tls) -> Result<(), AsyncQueueError> {
+        let manager = PostgresConnectionManager::new_from_stringlike(self.uri.clone(), tls)?;
 
-        Ok(Self {
-            pool,
-            duplicated_tasks,
-        })
-    }
+        let pool = Pool::builder()
+            .max_size(self.number_conns)
+            .build(manager)
+            .await?;
 
+        self.pool = Some(pool);
+        self.connected = true;
+        Ok(())
+    }
+    pub fn get_number_conns(&self) -> u32 {
```

Comment:
you're using this method only in the worker pool. But I think it's fine to maintain separate values for the number of workers and the number of connections

because the same queue can be used not only by workers but also in user code.

Reply:
Solve in [20e4757](https://github.com/ayrat555/fang/pull/54/commits/20e4757e6412d6819109d60b1ab308ded0863332)